src/cmd/flux-job.c

```diff
@@ -3463,6 +3463,61 @@ void info_output_jobspec (flux_future_t *f, struct info_ctx *ctx)
     }
 }
 
+void info_output_R (flux_future_t *f, struct info_ctx *ctx)
+{
+    if (ctx->base) {
+        const char *R_str;
+        info_output_get (f, ctx, "R", &R_str);
+        printf ("%s\n", R_str);
+    }
+    else {
+        const char *R_str;
+        const char *eventlog_str;
+        json_t *R;
+        json_t *eventlog;
+        json_error_t error;
+        size_t index;
+        json_t *entry;
+        char *R_updated;
+
+        info_output_get (f, ctx, "R", &R_str);
+        info_output_get (f, ctx, "eventlog", &eventlog_str);
+
+        if (!(R = json_loads (R_str, JSON_DECODE_ANY, &error)))
+            log_msg_exit ("Failed to decode R: %s", error.text);
+
+        if (!(eventlog = eventlog_decode (eventlog_str)))
+            log_err_exit ("Failed to decode eventlog");
+
+        json_array_foreach (eventlog, index, entry) {
+            const char *name;
+            json_t *context;
+            const char *path;
+            json_t *value;
+
+            if (eventlog_entry_parse (entry, NULL, &name, &context) < 0)
+                log_err_exit ("Failed to parse eventlog entry");
+
+            if (!streq (name, "resource-update"))
+                continue;
+
+            json_object_foreach (context, path, value) {
+                if (jpath_set (R, path, value) < 0)
+                    log_err_exit ("Failed to update R");
+            }
```

Comment:
RFC 21 specifies that a [resource-update](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_21.html#resource-update-event) event may only have a key of `expiration`, which updates the resource set expiration. I don't think this code will work here since the expiration is located in `execution.expiration`.

In the future other keys may indicate grow or shrink (or those may be different events) and those would likely not apply with `jpath_set()` either, so probably better to specifically handle `expiration` here and generate an error for any other key in the `resource-update` event context.